frontend/dockerfile/builder/build.go

```diff
@@ -196,12 +194,27 @@ func filter(opt map[string]string, key string) map[string]string {
 	return m
 }
 
-func parseGitSource(ref string) llb.State {
-	ref = strings.TrimPrefix(ref, gitPrefix)
+func detectGitContext(ref string) (*llb.State, bool) {
+	found := false
+	if httpPrefix.MatchString(ref) && gitUrlPathWithFragmentSuffix.MatchString(ref) {
+		found = true
+	}
+
```

Comment:
This is not robust, although go get does similar stuff with support for gitlab and bitbucket (IIRC docker does not)

Can we use another opt string?

Reply:
`docker build` uses similar logic to switch between http/git. If we use a separate field we just need to do this in a wrapper and users need to remember that `docker build` and `dockerfile` frontend take different opts. Also, we probably still would need to validate this value in a similar way as `llb.Git()` doesn't necessarily handle any string.

Reply:
IIUC `docker build` does not depend on `"github.com"` string, unlike this PR and `go get`?
https://github.com/moby/moby/blob/master/builder/remotecontext/git/gitutils.go

Reply:
https://github.com/moby/moby/blob/master/pkg/urlutil/urlutil.go#L13 (this is called by the CLI)

Reply:
😮 sorry didn't notice

Reply:
Note that on the CLI, a test was added (https://github.com/docker/cli/pull/390) to make sure that an existing local directory starting with `github.com/` is not ignored; https://github.com/docker/cli/blob/0f11a310fd7d64a697d0ed9098205b6cdd585f1b/cli/command/image/build_test.go#L176-L196